runtime/src/system_instruction_processor.rs

```diff
@@ -57,26 +66,34 @@ fn allocate(
     address: &Address,
     space: u64,
     signers: &HashSet<Pubkey>,
+    invoke_context: &mut dyn InvokeContext,
 ) -> Result<(), InstructionError> {
     if !address.is_signer(signers) {
-        debug!("Allocate: must carry signature of `to`");
+        ic_msg!(
+            invoke_context,
+            "Allocate: 'to' account {:?} is not a signer",
```

Comment:
```suggestion
            "Allocate: 'to' account {:?} must sign",
```

I think we can be more direct with these